### PR TITLE
auth: add full option to 'pdns_control show-config'

### PR DESCRIPTION
**Pull request by mind04**
### Short description
Without the full option only modified options are included in the output.
In full mode the output has the same format as 'pdns_server --config'.

### Checklist
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/arguments.hh [hunk 1]
```diff
@@ -106,9 +106,11 @@ public:
   void setCmd(const string &, const string &); //!< Add a command flag
   string &setSwitch(const string &, const string &); //!< Add a command flag
   string helpstring(string prefix=""); //!< generates the --help
-  string configstring(bool current=false); //!< generates the --mkconfig
+  string configstring(bool current=false, bool full=true); //!< generates the --mkconfig
```
Comment (zeha) [inline pdns/arguments.hh]: the pre-existing comment here is... outdated?

**Comment by zeha** [comment]
orthogonal: for the debian init scripted i would have needed a way of asking the pdns_server binary (not a running instance) what the value of a config setting (socket-dir) would be (after applying defaults and any config files)...

**Comment by Habbie** [comment]
> orthogonal: for the debian init scripted i would have needed a way of asking the pdns_server binary (not a running instance) what the value of a config setting (socket-dir) would be (after applying defaults and any config files)...

I suggest making a separate ticket for that, as we're about to merge this PR and then nobody will ever see your comment again.

File: pdns/arguments.hh (hunk 1 above)
Comment (Habbie) [reply pdns/arguments.hh]: looks like this PR fixes that now

**Comment by Habbie** [comment]
> orthogonal: for the debian init scripted i would have needed a way of asking the pdns_server binary (not a running instance) what the value of a config setting (socket-dir) would be (after applying defaults and any config files)...

With this PR:
```
$ pdns/pdns_server --config=diff --config-dir=. | grep ^socket-dir
socket-dir=.
```

File: pdns/arguments.hh [hunk 2]
```diff
@@ -106,9 +106,11 @@ public:
   void setCmd(const string &, const string &); //!< Add a command flag
   string &setSwitch(const string &, const string &); //!< Add a switch flag
   string helpstring(string prefix=""); //!< generates the --help
-  string configstring(bool current=false); //!< generates the --mkconfig
+  string configstring(bool current=false, bool full=true); //!< generates the --config
```
Comment (zeha) [inline pdns/arguments.hh]: methods with multiple default boolean args are weird; should one of them maybe not have a default arg or this be split into two (perhaps three) methods?

Comment (Habbie) [reply pdns/arguments.hh]: I also wondered about that, perhaps it would be best to not give them defaults at all?

File: pdns/receiver.cc [hunk 3]
```diff
@@ -569,7 +587,7 @@ int main(int argc, char **argv)
     DynListener::registerFunc("REMOTES", &DLRemotesHandler, "get top remotes");
     DynListener::registerFunc("SET",&DLSettingsHandler, "set config variables", "<var> <value>");
     DynListener::registerFunc("RETRIEVE",&DLNotifyRetrieveHandler, "retrieve slave domain", "<domain>");
-    DynListener::registerFunc("CURRENT-CONFIG",&DLCurrentConfigHandler, "retrieve the current configuration");
+    DynListener::registerFunc("CURRENT-CONFIG",&DLCurrentConfigHandler, "retrieve the current configuration", "[diff|default]");
```
Comment (zeha) [inline pdns/receiver.cc]: i wonder if `default` is a useful thing to ask a running instance?

**Comment by Habbie** [comment]
Have the review comments been addressed? If so, can you mark them resolved? :)

File: pdns/arguments.hh (hunk 2 above)
Comment (mind04) [reply pdns/arguments.hh]: the defaults are gone

**Pull request merged**